controllers/storagecluster/initialization_reconciler_test.go

```diff
@@ -149,87 +149,6 @@ func createUpdateRuntimeObjects(t *testing.T) []client.Object {
 	return updateRTObjects
 }
 
-func initStorageClusterResourceCreateUpdateTestProviderMode(t *testing.T, runtimeObjs []client.Object,
-	customSpec *api.StorageClusterSpec, storageProfiles *api.StorageProfileList, remoteConsumers bool) (*testing.T, StorageClusterReconciler,
-	*api.StorageCluster, reconcile.Request, []reconcile.Request) {
-	cr := createDefaultStorageCluster()
-	if customSpec != nil {
-		_ = mergo.Merge(&cr.Spec, customSpec)
-	}
-	requestOCSInit := reconcile.Request{
-		NamespacedName: types.NamespacedName{
-			Name:      "ocsinit",
-			Namespace: "",
-		},
-	}
-	requests := []reconcile.Request{requestOCSInit}
-
-	rtObjsToCreateReconciler := []runtime.Object{&nbv1.NooBaa{}}
-	// runtimeObjs are present, it means tests are for update
-	// add all the update required changes
-	if runtimeObjs != nil {
-		tbd := &appsv1.Deployment{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "rook-ceph-tools",
-			},
-		}
-		rtObjsToCreateReconciler = append(rtObjsToCreateReconciler, tbd)
-	}
-
-	if remoteConsumers {
-
-		clientConfigMap := &v1.ConfigMap{
-			ObjectMeta: metav1.ObjectMeta{Name: ocsClientConfigMapName},
-		}
-
-		rtObjsToCreateReconciler = append(rtObjsToCreateReconciler, clientConfigMap)
-
-		util.AddAnnotation(cr, "ocs.openshift.io/deployment-mode", "provider")
-	}
-
-	// Unpacks StorageProfile list to runtime objects array
-	var storageProfileRuntimeObjects []runtime.Object
-	for _, sp := range storageProfiles.Items {
-		storageProfileRuntimeObjects = append(storageProfileRuntimeObjects, &sp)
-	}
-	rtObjsToCreateReconciler = append(rtObjsToCreateReconciler, storageProfileRuntimeObjects...)
-
-	reconciler := createFakeInitializationStorageClusterReconciler(t, rtObjsToCreateReconciler...)
-
-	_ = reconciler.Client.Create(context.TODO(), cr)
-	for _, rtObj := range runtimeObjs {
-		_ = reconciler.Client.Create(context.TODO(), rtObj)
-	}
-
-	var requestsStorageProfiles []reconcile.Request
-	for _, sp := range storageProfiles.Items {
-		requestSP := reconcile.Request{
-			NamespacedName: types.NamespacedName{
-				Name:      sp.Name,
-				Namespace: cr.Namespace,
-			},
-		}
-		_ = reconciler.Client.Create(context.TODO(), &sp)
-		requestsStorageProfiles = append(requestsStorageProfiles, requestSP)
-		requests = append(requests, requestSP)
-
-	}
-
-	spList := &api.StorageProfileList{}
-	_ = reconciler.Client.List(context.TODO(), spList)
-	for _, request := range requests {
-		err := os.Setenv("OPERATOR_NAMESPACE", request.Namespace)
-		assert.NoError(t, err)
-		result, err := reconciler.Reconcile(context.TODO(), request)
-		assert.NoError(t, err)
-		assert.Equal(t, reconcile.Result{}, result)
-		err = os.Setenv("WATCH_NAMESPACE", request.Namespace)
-		assert.NoError(t, err)
-	}
-
-	return t, reconciler, cr, requestOCSInit, requestsStorageProfiles
-}
-
```

Comment:
Why are we removing all of this and not just the storage profile parts?

Reply:
This is not being used anywhere else and it was created for storageprofile tests itself. you can refer to the 2nd commit in the description for the same.